### PR TITLE
Implement LocalRegistry

### PR DESCRIPTION
**Pull request by jmrossy**
- Create new LocalRegistry class
- Add new export entry for local-related things
- Refactor unit tests to cover both registry types

### CODE REVIEW

File: src/registry/LocalRegistry.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  constructor(options: LocalRegistryOptions) {
+    super({ logger: options.logger });
+    this.uri = options.uri;
+  }
+
+  listRegistryContent(): RegistryContent {
+    if (this.listContentCache) return this.listContentCache;
+
+    const chainFileList = this.listFiles(path.join(this.uri, this.getChainsPath()));
+    const chains: ChainMap<ChainFiles> = {};
```
Comment (jmrossy) [inline src/registry/LocalRegistry.ts:40]: There's some code duplication between this file and GithubRegistry but importantly, all the methods here are sync, which makes code reuse tricky. Given that it's <30 dupe lines total, I'm just going with this for now. We can revisit if more duplication arises when adding the write-related methods

File: test/unit/registry.test.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import { expect } from 'chai';
+
+import { GithubRegistry } from '../../src/registry/GithubRegistry.js';
+import { LocalRegistry } from '../../src/registry/LocalRegistry.js';
+
+describe('Registry utilities', () => {
```
Comment (jmrossy) [inline test/unit/registry.test.ts:6]: Not really new, this is a modified `GithubRegistry.test.ts` but I guess git didn't make the connection

File: src/registry/LocalRegistry.ts (hunk 1 above)
Comment (yorhodes) [reply src/registry/LocalRegistry.ts:35]: could you use your MaybePromise thing somehow?

File: src/registry/LocalRegistry.ts [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    const addresses = this.getAddresses();
+    if (!addresses[chainName])
+      throw new Error(`Addresses not found in registry for chain: ${chainName}`);
+    return addresses[chainName];
+  }
+
+  protected listFiles(dirPath: string): string[] {
+    let entries = fs.readdirSync(dirPath, { withFileTypes: true });
+
+    let filePaths = entries.map((entry) => {
```
Comment (yorhodes) [inline src/registry/LocalRegistry.ts:95]: nit: const?

File: src/utils.ts
Comment (yorhodes) [inline src/utils.ts:1]: rm?

File: src/registry/LocalRegistry.ts (hunk 1 above)
Comment (yorhodes) [reply src/registry/LocalRegistry.ts:35]: having all of the per/multi chain stuff be reused might be nice

**Review by yorhodes — APPROVED** [review]

File: src/registry/LocalRegistry.ts (hunk 3 above)
Comment (jmrossy) [reply src/registry/LocalRegistry.ts:95]: I've set up eslint to catch these, thanks

**Pull request merged**